client/client.go

```diff
@@ -269,3 +309,16 @@ func GetExponentialBackoffTime(tried int, maxInterval time.Duration) (time.Durat
 
 	return interval, nil
 }
+
+// unmarshalErrorMessage unmarshals the error message contained in an
+// error request from the server.
+func unmarshalErrorMessage(r io.Reader) string {
+	e := new(struct {
+		Error string `json:"error"`
+		ReqID string `json:"request_id"`
+	})
+	if err := json.NewDecoder(r).Decode(e); err != nil {
+		return fmt.Sprintf("failed to parse server response: %v", err)
+	}
+	return fmt.Sprintf("request (id: %s) error: %s", e.ReqID, e.Error)
```

Comment:
Does it mean we are printing sometimes `request_id` twice in one error message? One will be added here: https://github.com/mendersoftware/mender/pull/271/files#diff-1cd41a2611da49ce63630d5a51bd82c5R96 and one as the part of unmarshalling.

Reply:
ur right. nice find.